src/api/spec/models/user_spec.rb

```diff
@@ -623,4 +623,48 @@
       end
     end
   end
+
+  describe '#can_modify?' do
+    let(:user) { create(:confirmed_user, :with_home, login: 'hans') }
+    let(:project) { user.home_project }
+
+    context 'when dealing with projects' do
+      context 'when checking against a home project' do
+        it { expect(user.can_modify?(project)).to be true }
+      end
+
+      context 'when checking against a child of a home project' do
+        let(:child_project) { create(:project, name: "#{project.name}:something") }
+
+        before do
+          child_project.relationships.build(user: user,
+                                            role: Role.find_by_title('maintainer'))
+        end
+
+        it { expect(user.can_modify?(child_project)).to be true }
+      end
+
+      context 'when the project has no parent project' do
+        let(:user) { create(:confirmed_user) }
+
+        context 'but the user is the root owner' do
+          let(:child_project) { create(:project, name: "home:#{user.login}:something") }
+
+          it { expect(user.can_modify?(child_project)).to be true }
+        end
+      end
+
+      context 'when checking other users home projects' do
+        let(:other_user) { create(:confirmed_user, :with_home, login: 'hans2') }
+
+        it { expect(user.can_modify?(other_user.home_project)).to be false }
+      end
+
+      context 'when checking branched projects' do
```

Comment:
Not sure why you are testing this?

Reply:
To ensure permissions don't allow modifying other people's projects